src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs

```diff
@@ -437,6 +437,18 @@ private IEnumerable<OutputLibFile> InitLibFiles(IMSBuildItem[] libFiles)
                     throw new PackagingException(NuGetLogCode.NU5027, string.Format(CultureInfo.CurrentCulture, Strings.InvalidTargetFramework, finalOutputPath));
                 }
 
+                if (!string.IsNullOrEmpty(targetFramework))
+                {
+                    var fw = NuGetFramework.Parse(targetFramework);
+                    if (!string.IsNullOrEmpty(fw.Platform) && fw.PlatformVersion == FrameworkConstants.EmptyVersion)
```

Comment:
>!string.IsNullOrEmpty(fw.Platform) && fw.PlatformVersion == FrameworkConstants.EmptyVersion [](start = 24, length = 91)

Just for my understanding, is there a case where having a PlatformVersion without a Platform is valid?

Reply:
nope!

Reply:
In that case, do we need the first check for fw.Platform?

Reply:
Oh, yes we do, because "PlatformVersion" will me EmptyVersion when Platform is empty, and we only wanna do this check when we definitely have a platform.